src/main/java/seedu/address/model/tag/Tag.java

```diff
@@ -32,6 +32,39 @@ public static boolean isValidTagName(String test) {
         return test.matches(VALIDATION_REGEX);
     }
 
+    /**
+     * Returns true if another tag has the same TagName as this tag.
+     * @param otherTag A tag to compare with.
+     */
+    public boolean isSameTag(Tag otherTag) {
+        if (otherTag == this) {
+            return true;
+        }
+
+        return otherTag != null
+                && otherTag.getTagName().equals(getTagName());
+    }
+
+    public TagName getTagName() { return tagName; }
+
+    public int getNumberOfPersonsTagged() { return taggedCount; }
+
+    public void increaseTaggedCount() {
+        taggedCount++;
+    }
+
+    public void decreaseTaggedCount() {
+        taggedCount--;
+    }
+
+    /**
+     * Returns true if the tag can be deleted.
+     * The tag can be deleted if TaggedCount is 0.
+     */
+    public boolean canBeDeleted() {
```

Comment:
Hmm interesting approach I thought if a tag was deleted from WedLinker then all contacts would immediately lose that tag... seems like this forces the user to remove all tags from the contacts having this tag before it can be deleted?

Reply:
I thought it was logical to ensure the tag is not being used but good suggestion. Would implement this feature after merging with Danza so that I can use the untag function to untag all contacts with the tag to be deleted.